server/src/main/java/io/crate/analyze/AnalyzedTableElements.java

```diff
@@ -119,18 +119,25 @@ private AnalyzedTableElements(List<AnalyzedColumnDefinition<T>> partitionedByCol
         this.copyToMap = copyToMap;
     }
 
+    public static void finalizeIndices(AnalyzedTableElements<Object> elements) {
+        for (AnalyzedColumnDefinition<Object> column : elements.columns) {
+            if (column.isIndexColumn()) {
+                elements.indicesMap.put(column.name(), column.toMetaIndicesMapping());
+            }
+        }
+    }
+
     public static Map<String, Object> toMapping(AnalyzedTableElements<Object> elements) {
         final Map<String, Object> mapping = new HashMap<>();
         final Map<String, Object> meta = new HashMap<>();
         final Map<String, Object> properties = new HashMap<>(elements.columns.size());
 
         Map<String, String> generatedColumns = new HashMap<>();
 
+        finalizeIndices(elements);
+
         for (AnalyzedColumnDefinition<Object> column : elements.columns) {
             properties.put(column.name(), AnalyzedColumnDefinition.toMapping(column));
-            if (column.isIndexColumn()) {
-                elements.indicesMap.put(column.name(), column.toMetaIndicesMapping());
-            }
             addToGeneratedColumns("", column, generatedColumns);
```

Comment:
Everything in toMapping which used to enrich AnalzyedTableElements fields or do some validation has to be taken out as we don't use toMapping in 5.4 and will remove it in the future. 

Primary keys are also added in `toMapping` (table level keys are added to primaryKeys) but we don't need to take it out since `finalizeAndvalidate` calls `validatePrimaryKeys` which enriches `AnalyzedTableElements` with additional PK-s.